src/http_client/request.rs

```diff
@@ -0,0 +1,83 @@
+use std::{fmt, future::Future, pin::Pin};
+
+use anyhow::{anyhow, Error};
+use futures_util::TryFutureExt;
+use http::{request::Builder, HeaderName, HeaderValue};
+use hyper::{body::Bytes, Body, Response};
+
+use super::client::Client;
+
+pub struct RequestBuilder {
+    client: Client,
+    req_builder: Builder,
+}
+
+impl RequestBuilder {
+    pub fn new(client: Client, req_builder: Builder) -> Self {
+        Self {
+            client,
+            req_builder,
+        }
+    }
+
+    pub fn header<K, V>(mut self, key: K, value: V) -> RequestBuilder
+    where
+        HeaderName: TryFrom<K>,
+        <HeaderName as TryFrom<K>>::Error: Into<http::Error>,
+        HeaderValue: TryFrom<V>,
+        <HeaderValue as TryFrom<V>>::Error: Into<http::Error>,
+    {
+        self.req_builder = self.req_builder.header(key, value);
+        self
+    }
+
+    pub async fn send(self) -> Result<Response<Body>, Error> {
+        let req = self
+            .req_builder
+            .header(http::header::USER_AGENT, "fluvio-mini-http/0.1")
```

Comment:
is this just for simple get?  then let's just move to read only

Reply:
I'm not sure what you meant by "read only" here

But I prefer to keep the `RequestBuilder` the way it is:
- it's closer to the `reqwest` API
- it's more extensible if we want to add new features
- it has better usability (it's not possible to `Client::get().get().get()...`, for example)

Reply:
k